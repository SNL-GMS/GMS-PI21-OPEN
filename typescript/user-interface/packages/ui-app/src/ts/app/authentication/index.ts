import { ianAuthenticator } from './ian-authentication';
import type { Authenticator } from './types';

export const authenticator: Authenticator = ianAuthenticator;
