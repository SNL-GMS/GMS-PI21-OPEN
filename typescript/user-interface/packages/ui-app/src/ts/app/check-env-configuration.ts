import {
  CESIUM_OFFLINE,
  GATEWAY_HTTP_PROXY_URI,
  GMS_UI_MODE,
  IS_MODE_IAN,
  IS_MODE_LEGACY,
  IS_MODE_SOH,
  IS_NODE_ENV_DEVELOPMENT,
  IS_NODE_ENV_PRODUCTION,
  NODE_ENV,
  SUBSCRIPTIONS_PROXY_URI,
  UI_URL
} from '@gms/common-util';
import { UILogger } from '@gms/ui-util';

import {
  GMS_DISABLE_KEYCLOAK_AUTH,
  GMS_KEYCLOAK_CLIENT_ID,
  GMS_KEYCLOAK_REALM,
  GMS_KEYCLOAK_URL
} from '~env';

const logger = UILogger.create('GMS_LOG_CONFIG', process.env.GMS_LOG_CONFIG);

/**
 * Checks the ENV Configuration.
 * Logs the current environment configuration.
 */
export const checkEnvConfiguration = (): void => {
  logger.debug(
    `Environment (process.env): ` +
      `\n   process.env.NODE_ENV=${process.env.NODE_ENV}` +
      `\n   process.env.GMS_UI_MODE=${process.env.GMS_UI_MODE}` +
      `\n   process.env.GATEWAY_HTTP_PROXY_URI=${process.env.GATEWAY_HTTP_PROXY_URI}` +
      `\n   process.env.WAVEFORMS_PROXY_URI=${process.env.WAVEFORMS_PROXY_URI}` +
      `\n   process.env.SUBSCRIPTIONS_PROXY_URI=${process.env.SUBSCRIPTIONS_PROXY_URI}` +
      `\n   process.env.CESIUM_OFFLINE=${process.env.CESIUM_OFFLINE}`
  );

  logger.debug(
    `App Environment: ` +
      `\n   NODE_ENV=${NODE_ENV}` +
      `\n   GMS_UI_MODE=${GMS_UI_MODE}` +
      `\n   IS_MODE_LEGACY=${IS_MODE_LEGACY}` +
      `\n   IS_MODE_IAN=${IS_MODE_IAN}` +
      `\n   IS_MODE_SOH=${IS_MODE_SOH}` +
      `\n   IS_NODE_ENV_DEVELOPMENT=${IS_NODE_ENV_DEVELOPMENT}` +
      `\n   IS_NODE_ENV_PRODUCTION=${IS_NODE_ENV_PRODUCTION}` +
      `\n   GATEWAY_HTTP_PROXY_URI=${GATEWAY_HTTP_PROXY_URI}` +
      `\n   SUBSCRIPTIONS_PROXY_URI=${SUBSCRIPTIONS_PROXY_URI}` +
      `\n   UI_URL=${UI_URL}` +
      `\n   CESIUM_OFFLINE=${CESIUM_OFFLINE}` +
      `\n   GMS_KEYCLOAK_URL=${GMS_KEYCLOAK_URL}` +
      `\n   GMS_KEYCLOAK_CLIENT_ID=${GMS_KEYCLOAK_CLIENT_ID}` +
      `\n   GMS_KEYCLOAK_REALM=${GMS_KEYCLOAK_REALM}` +
      `\n   GMS_DISABLE_KEYCLOAK_AUTH=${GMS_DISABLE_KEYCLOAK_AUTH}`
  );
};
