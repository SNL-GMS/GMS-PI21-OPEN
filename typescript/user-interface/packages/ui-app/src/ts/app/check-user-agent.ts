import { UILogger } from '@gms/ui-util';

const logger = UILogger.create('GMS_LOG_USER_AGENT', process.env.GMS_LOG_USER_AGENT);
/**
 * Checks the user agent.
 */
export const checkUserAgent = (): void => {
  if (navigator) {
    logger.info(`client connected ${navigator.userAgent.toLowerCase()}`);
  }

  if (
    !window.navigator.userAgent.includes('Chrome') &&
    !window.navigator.userAgent.includes('Firefox')
  ) {
    // eslint-disable-next-line no-alert
    window.alert(`GMS Interactive Analysis currently supports
            Google Chrome > v59 and Firefox > v. You will likely experience degraded performance`);
  }
};
