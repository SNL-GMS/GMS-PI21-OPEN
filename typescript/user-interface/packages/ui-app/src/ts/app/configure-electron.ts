import { getElectron } from '@gms/ui-util';

/**
 * Configures Electron.
 */
export const configureElectron = (): void => {
  // electron instance; undefined if not running in electron
  const electron = getElectron();
  if (electron !== undefined && electron.ipcRenderer !== undefined) {
    electron.ipcRenderer.on('load-path', (event, newHash: string) => {
      window.location.hash = newHash;
    });
  }
};
