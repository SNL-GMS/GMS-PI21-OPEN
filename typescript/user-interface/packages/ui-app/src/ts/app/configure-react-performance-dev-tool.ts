import { IS_NODE_ENV_DEVELOPMENT } from '@gms/common-util';

/**
 * Configures the React Performance Dev Tool.
 * ONLY ENABLED FOR NODE_ENV === 'development'
 */
export const configureReactPerformanceDevTool = (): void => {
  // REACT Performance DEV Tool, enabled only in develop
  if (IS_NODE_ENV_DEVELOPMENT) {
    // eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports, global-require
    const { registerObserver } = require('react-perf-devtool');
    const options = {};

    const callback = () => {
      /* no-op */
    };

    // assign the observer to the global scope, as the GC will delete it otherwise
    if (window) {
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      (window as any).observer = registerObserver(options, callback);
    }
  }
};
