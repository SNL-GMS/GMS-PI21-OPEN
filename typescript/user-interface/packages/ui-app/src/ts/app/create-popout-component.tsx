import { Colors } from '@blueprintjs/core';
import { GMS_UI_MODE } from '@gms/common-util';
import { getElectron } from '@gms/ui-util';
import React from 'react';

// electron instance; undefined if not running in electron
const electron = getElectron();

/**
 * Wrap the component with everything it needs to live standalone as a popout
 */
export const createPopoutComponent = (
  Component: any,
  props: any,
  suppressPopinIcon = false
): JSX.Element => {
  // eslint-disable-next-line react/display-name
  const PopoutComponent = class extends React.PureComponent {
    public render(): JSX.Element {
      return (
        <div
          style={{
            width: '100%',
            height: '100%',
            backgroundColor: Colors.DARK_GRAY2
          }}
          className="popout-component"
          data-gms-ui-mode={GMS_UI_MODE}
        >
          {/* eslint-disable-next-line react/jsx-props-no-spreading */}
          <Component {...this.props} />
          {
            // only show pop-in button if running in electron
            !suppressPopinIcon &&
            electron &&
            electron !== undefined &&
            electron.ipcRenderer !== undefined ? (
              // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
              <div
                className="lm_popin"
                title="pop-in"
                onClick={() => {
                  electron.ipcRenderer.send(
                    'popin-window',
                    electron.remote.getCurrentWebContents().popoutConfig
                  );
                  electron.remote.getCurrentWindow().close();
                }}
              >
                <div className="lm_icon" />
                <div className="lm_bg" />
              </div>
            ) : undefined
          }
        </div>
      );
    }
  };

  // eslint-disable-next-line react/jsx-props-no-spreading
  return <PopoutComponent {...props} />;
};
