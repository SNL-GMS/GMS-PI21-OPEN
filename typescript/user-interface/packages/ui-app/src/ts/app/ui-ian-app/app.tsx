import { Intent, NonIdealState, Spinner } from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import { Displays } from '@gms/common-model';
import { IanDisplays } from '@gms/common-model/lib/displays/types';
import { IS_MODE_IAN } from '@gms/common-util';
import { getStore, useUiTheme, withReduxProvider } from '@gms/ui-state';
import * as React from 'react';
import { Provider } from 'react-redux';
import { HashRouter, Route, Routes } from 'react-router-dom';

import { AnalystUiComponents } from '~analyst-ui/';
import {
  IANMap,
  SignalDetectionsComponent,
  StationPropertiesComponent
} from '~analyst-ui/components';
import { FilterManager } from '~analyst-ui/components/filters/filter-manager';
import { InteractionConsumer } from '~analyst-ui/interactions/interaction-consumer';
import { InteractionProvider } from '~analyst-ui/interactions/interaction-provider';
import { authenticator } from '~app/authentication';
import { EffectiveNowTimeInitializer } from '~app/initializers/effective-time-now-initializer';
import { UIThemeWrapper } from '~app/initializers/ui-theme-wrapper';
import { ServiceWorkerController } from '~app/service-worker-controller';
import { ThemedToastContainer } from '~app/themed-toast-container';
import { CommonCommandRegistrar } from '~components/common-ui/commands';
import { CommandPalette } from '~components/common-ui/components/command-palette';
import { LoadingScreen } from '~components/loading-screen';
import { LoginScreen } from '~components/login-screen';
import { ProtectedRoute } from '~components/protected-route';
import { GoldenLayoutContext, Workspace } from '~components/workspace';
import { GMS_DISABLE_KEYCLOAK_AUTH } from '~env';

import { createPopoutComponent } from '../create-popout-component';
import { glContextData } from './golden-layout-config';

const store = getStore();

/**
 * Wraps the component route (not for SOH).
 * Provides the required context providers to the component.
 *
 * @param Component the component route
 * @param props the props passed down from the route to the component
 * @param suppressPopinIcon true to force suppress the golden-layout popin icon
 */
function wrap(Component: any, props: any, suppressPopinIcon = false) {
  function WrappedComponent(p) {
    const [uiTheme] = useUiTheme();
    return (
      <ServiceWorkerController>
        <>
          <UIThemeWrapper key={uiTheme.name}>
            <CommandPalette>
              {/* eslint-disable-next-line react/jsx-props-no-spreading */}
              <Component {...p} />
              <InteractionProvider>
                <InteractionConsumer />
              </InteractionProvider>
              <CommonCommandRegistrar />
            </CommandPalette>
            <ThemedToastContainer />
          </UIThemeWrapper>
          <EffectiveNowTimeInitializer />
          <FilterManager />
        </>
      </ServiceWorkerController>
    );
  }

  return createPopoutComponent(withReduxProvider(WrappedComponent), props, suppressPopinIcon);
}

const WrappedWorkflow: React.FC<unknown> = props => wrap(AnalystUiComponents.Workflow, props);
const WrappedEvents: React.FC<unknown> = props => wrap(AnalystUiComponents.Events, props);
const WrappedFilters: React.FC<unknown> = props => wrap(AnalystUiComponents.Filters, props);
const WrappedWaveformDisplay: React.FC<unknown> = props =>
  wrap(AnalystUiComponents.WaveformDisplay, props);
const WrappedMap: React.FC<unknown> = props => wrap(IANMap, props);
const WrappedSignalDetections: React.FC<unknown> = props => wrap(SignalDetectionsComponent, props);
const WrappedStationProperties: React.FC<unknown> = props =>
  wrap(StationPropertiesComponent, props);
const WrappedWorkspace: React.FC<unknown> = props => wrap(Workspace, props, true);

export function App(): React.ReactElement {
  // TODO: either re-route or make this a <404 component>
  const redirectRoute = GMS_DISABLE_KEYCLOAK_AUTH ? (
    <>
      <Route path="/login/:redirectUrl" element={<LoginScreen authenticator={authenticator} />} />
      <Route path="/login" element={<LoginScreen authenticator={authenticator} />} />
    </>
  ) : (
    <Route path="/:redirectUrl" />
  );

  return !IS_MODE_IAN ? (
    <NonIdealState
      icon={IconNames.ERROR}
      action={<Spinner intent={Intent.DANGER} />}
      title="Invalid settings"
      description="Not configured for IAN mode - Please check settings"
    />
  ) : (
    <Provider store={store}>
      <HashRouter>
        {
          // ! CAUTION: when changing the route paths
          // The route paths must match the `golden-layout` component name for popout windows
          // For example, the component name `my-route` must have the route path of `my-route`
        }
        {
          // For performance use `render` which accepts a functional component
          // that won't get unnecessarily remounted like with component.
        }
        <Routes>
          {
            // Authentication
            // React router dropped support for optional params
            // so we need separate routes for login and login with a redirect
          }
          {redirectRoute}
          <Route
            path="/loading"
            element={
              <ProtectedRoute redirectPath="loading" title="GMS: Loading...">
                <LoadingScreen />
              </ProtectedRoute>
            }
          />
          {
            // Individual ProtectedRoutes
          }
          <Route
            path={`/${IanDisplays.WORKFLOW}`}
            element={
              <ProtectedRoute
                redirectPath={IanDisplays.WORKFLOW}
                title={Displays.toDisplayTitle(Displays.IanDisplays.WORKFLOW, 'GMS: ')}
              >
                <WrappedWorkflow />
              </ProtectedRoute>
            }
          />
          <Route
            path={`/${IanDisplays.EVENTS}`}
            element={
              <ProtectedRoute
                redirectPath={IanDisplays.EVENTS}
                title={Displays.toDisplayTitle(Displays.IanDisplays.EVENTS, 'GMS: ')}
              >
                <WrappedEvents />
              </ProtectedRoute>
            }
          />
          <Route
            path={`/${IanDisplays.FILTERS}`}
            element={
              <ProtectedRoute
                redirectPath={IanDisplays.FILTERS}
                title={Displays.toDisplayTitle(Displays.IanDisplays.FILTERS, 'GMS: ')}
              >
                <WrappedFilters />
              </ProtectedRoute>
            }
          />
          <Route
            path={`/${IanDisplays.WAVEFORM}`}
            element={
              <ProtectedRoute
                redirectPath={IanDisplays.WAVEFORM}
                title={Displays.toDisplayTitle(Displays.IanDisplays.WAVEFORM, 'GMS: ')}
              >
                <WrappedWaveformDisplay />
              </ProtectedRoute>
            }
          />
          <Route
            path={`/${IanDisplays.MAP}`}
            element={
              <ProtectedRoute
                redirectPath={IanDisplays.MAP}
                title={Displays.toDisplayTitle(Displays.IanDisplays.MAP, 'GMS: ')}
              >
                <WrappedMap />
              </ProtectedRoute>
            }
          />
          <Route
            path={`/${IanDisplays.SIGNAL_DETECTIONS}`}
            element={
              <ProtectedRoute
                redirectPath={IanDisplays.SIGNAL_DETECTIONS}
                title={Displays.toDisplayTitle(Displays.IanDisplays.SIGNAL_DETECTIONS, 'GMS: ')}
              >
                <WrappedSignalDetections />
              </ProtectedRoute>
            }
          />
          <Route
            path={`/${IanDisplays.STATION_PROPERTIES}`}
            element={
              <ProtectedRoute
                redirectPath={IanDisplays.STATION_PROPERTIES}
                title={Displays.toDisplayTitle(Displays.IanDisplays.STATION_PROPERTIES, 'GMS: ')}
              >
                <WrappedStationProperties />
              </ProtectedRoute>
            }
          />
          {
            // Workspace
          }
          <Route
            path="*"
            element={
              <ProtectedRoute title="GMS: App Workspace">
                <GoldenLayoutContext.Provider value={glContextData()}>
                  <WrappedWorkspace />
                </GoldenLayoutContext.Provider>
              </ProtectedRoute>
            }
          />
        </Routes>
      </HashRouter>
    </Provider>
  );
}
