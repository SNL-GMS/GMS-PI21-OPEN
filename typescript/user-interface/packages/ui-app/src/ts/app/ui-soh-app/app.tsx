import { Intent, NonIdealState, Spinner } from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import { IS_MODE_SOH } from '@gms/common-util';
import {
  getStore,
  SohStatusSubscription,
  SystemMessageSubscription,
  withReduxProvider
} from '@gms/ui-state';
import * as React from 'react';
import { Provider } from 'react-redux';
import { HashRouter, Route, Routes } from 'react-router-dom';

import { authenticator } from '~app/authentication';
import { UIThemeWrapper } from '~app/initializers/ui-theme-wrapper';
import { ThemedToastContainer } from '~app/themed-toast-container';
import { SystemMessageAlertPlayer } from '~common-ui/components/system-message/audio';
import { CommonUIComponents } from '~components/common-ui';
import { CommonCommandRegistrar } from '~components/common-ui/commands';
import { CommandPalette } from '~components/common-ui/components/command-palette';
import { SohCommandRegistrar } from '~components/data-acquisition-ui/commands';
import { SohEnvironmentHistory } from '~components/data-acquisition-ui/components/environment-history';
import { SohLagHistory } from '~components/data-acquisition-ui/components/soh-lag-history';
import { SohMissingHistory } from '~components/data-acquisition-ui/components/soh-missing-history';
import { SohTimelinessHistory } from '~components/data-acquisition-ui/components/soh-timeliness-history';
import { StationStatistics } from '~components/data-acquisition-ui/components/station-statistics';
import { InteractionWrapper } from '~components/data-acquisition-ui/interactions/interaction-wrapper';
import { LoadingScreen } from '~components/loading-screen';
import { LoginScreen } from '~components/login-screen';
import { ProtectedRoute } from '~components/protected-route';
import { GoldenLayoutContext, Workspace } from '~components/workspace';
import { SohLag, SohMissing, SohTimeliness } from '~data-acquisition-ui/components/soh-bar-chart';
import { SohEnvironment } from '~data-acquisition-ui/components/soh-environment';
import { SohMap } from '~data-acquisition-ui/components/soh-map';
import { SohOverview } from '~data-acquisition-ui/components/soh-overview';

import { createPopoutComponent } from '../create-popout-component';
import { glContextData } from './golden-layout-config';

/**
 * Wraps the component route (not for SOH).
 * Provides the required context providers to the component.
 *
 * @param Component the component route
 * @param props the props passed down from the route to the component
 * @param suppressPopinIcon true to force suppress the golden-layout popin icon
 */
function wrap(Component: any, props: any, suppressPopinIcon = false) {
  const InteractionComp = InteractionWrapper(Component);

  function WrappedComponent(p) {
    return (
      <CommandPalette>
        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
        <InteractionComp {...p} />
        <CommonCommandRegistrar />
        <ThemedToastContainer />
        <SystemMessageAlertPlayer />
      </CommandPalette>
    );
  }

  return createPopoutComponent(
    withReduxProvider(
      SystemMessageSubscription.wrapSystemMessageSubscription(WrappedComponent, props)
    ),
    props,
    suppressPopinIcon
  );
}

/**
 * Wraps the component route for SOH.
 * Provides the required context providers to the component.
 *
 * @param Component the component route
 * @param props the props passed down from the route to the component
 * @param suppressPopinIcon true to force suppress the golden-layout popin icon
 */
function wrapSoh(Component: any, props: any, suppressPopinIcon = false) {
  const InteractionComp = InteractionWrapper(Component);

  function WrappedComponent(p) {
    return (
      <UIThemeWrapper>
        <CommandPalette>
          {/* eslint-disable-next-line react/jsx-props-no-spreading */}
          <InteractionComp {...p} />
          <SohCommandRegistrar />
          <CommonCommandRegistrar />
        </CommandPalette>
        <ThemedToastContainer />
        <SystemMessageAlertPlayer />
      </UIThemeWrapper>
    );
  }
  return createPopoutComponent(
    withReduxProvider(
      SystemMessageSubscription.wrapSystemMessageSubscription(
        SohStatusSubscription.wrapSohStatusSubscription(WrappedComponent, props),
        props
      )
    ),
    props,
    suppressPopinIcon
  );
}

const WrappedSystemMessage: React.FC = props => wrap(CommonUIComponents.SystemMessage, props);
const WrappedOverview: React.FC = props => wrapSoh(SohOverview, props);
const WrappedStatistics: React.FC = props => wrapSoh(StationStatistics, props);
const WrappedEnvironment: React.FC = props => wrapSoh(SohEnvironment, props);
const WrappedMissing: React.FC = props => wrapSoh(SohMissing, props);
const WrappedLag: React.FC = props => wrapSoh(SohLag, props);
const WrappedEnvironmentTrends: React.FC = props => wrapSoh(SohEnvironmentHistory, props);
const WrappedLagTrends: React.FC = props => wrapSoh(SohLagHistory, props);
const WrappedMissingTrends: React.FC = props => wrapSoh(SohMissingHistory, props);
const WrappedTimelinessTrends: React.FC = props => wrapSoh(SohTimelinessHistory, props);
const WrappedTimeliness: React.FC = props => wrapSoh(SohTimeliness, props);
const WrappedMap: React.FC = props => wrapSoh(SohMap, props);
const WrappedWorkspace: React.FC = props => wrapSoh(Workspace, props, true);

export function App(): React.ReactElement {
  return !IS_MODE_SOH ? (
    <NonIdealState
      icon={IconNames.ERROR}
      action={<Spinner intent={Intent.DANGER} />}
      title="Invalid settings"
      description="Not configured for SOH mode - Please check settings"
    />
  ) : (
    <Provider store={getStore()}>
      <HashRouter>
        {
          // ! CAUTION: when changing the route paths
          // The route paths must match the `golden-layout` component name for popout windows
          // For example, the component name `signal-detections` must have the route path of `signal-detections`
        }
        {
          // For performance use `render` which accepts a functional component
          // that won't get unnecessarily remounted like with component.
        }
        <Routes>
          {
            // Authentication
            // React router dropped support for optional params
            // so we need separate routes for login and login with a redirect
          }
          <Route
            path="/login/:redirectUrl"
            element={<LoginScreen authenticator={authenticator} />}
          />
          <Route path="/login" element={<LoginScreen authenticator={authenticator} />} />
          <Route
            path="/loading"
            element={
              <ProtectedRoute>
                <LoadingScreen />
              </ProtectedRoute>
            }
          />
          {
            // Common UI
          }
          <Route
            path="/system-messages"
            element={
              <ProtectedRoute>
                <WrappedSystemMessage />
              </ProtectedRoute>
            }
          />
          {
            // Data Acquisition
          }
          <Route
            path="/soh-overview"
            element={
              <ProtectedRoute>
                <WrappedOverview />
              </ProtectedRoute>
            }
          />
          <Route
            path="/station-statistics"
            element={
              <ProtectedRoute>
                <WrappedStatistics />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-environment"
            element={
              <ProtectedRoute>
                <WrappedEnvironment />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-missing"
            element={
              <ProtectedRoute>
                <WrappedMissing />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-lag"
            element={
              <ProtectedRoute>
                <WrappedLag />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-environment-trends"
            element={
              <ProtectedRoute>
                <WrappedEnvironmentTrends />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-lag-trends"
            element={
              <ProtectedRoute>
                <WrappedLagTrends />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-missing-trends"
            element={
              <ProtectedRoute>
                <WrappedMissingTrends />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-timeliness-trends"
            element={
              <ProtectedRoute>
                <WrappedTimelinessTrends />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-timeliness"
            element={
              <ProtectedRoute>
                <WrappedTimeliness />
              </ProtectedRoute>
            }
          />
          <Route
            path="/soh-map"
            element={
              <ProtectedRoute>
                <WrappedMap />
              </ProtectedRoute>
            }
          />
          {
            // Workspace
          }
          <Route
            path="*"
            element={
              <ProtectedRoute>
                <GoldenLayoutContext.Provider value={glContextData()}>
                  <WrappedWorkspace />
                </GoldenLayoutContext.Provider>
              </ProtectedRoute>
            }
          />
        </Routes>
      </HashRouter>
    </Provider>
  );
}
