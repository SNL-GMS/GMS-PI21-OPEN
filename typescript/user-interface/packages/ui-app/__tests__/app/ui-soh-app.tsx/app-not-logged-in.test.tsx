import { Logger } from '@gms/common-util';
import { act, render } from '@testing-library/react';
import * as React from 'react';

import { App } from '../../../src/ts/app/ui-soh-app/app';

const logger = Logger.create('GMS_LOG_JEST', process.env.GMS_LOG_JEST);

jest.mock('../../../src/ts/components/data-acquisition-ui/components/soh-map', () => {
  return { SOHMap: () => logger.debug('map') };
});

jest.mock('@gms/common-util', () => {
  const actual = jest.requireActual('@gms/common-util');
  return {
    ...actual,
    IS_MODE_SOH: true,
    IS_MODE_IAN: false,
    IS_MODE_LEGACY: false
  };
});

describe('SOH App - Not Logged In', () => {
  // TODO Unskip and fix
  it('matches a snapshot', () => {
    act(() => {
      const { container } = render(<App />);
      expect(container).toMatchSnapshot();
    });
  });
});
