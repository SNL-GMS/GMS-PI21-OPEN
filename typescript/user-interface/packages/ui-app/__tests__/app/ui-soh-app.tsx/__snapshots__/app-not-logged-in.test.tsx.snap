// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SOH App - Not Logged In matches a snapshot 1`] = `<div />`;
